applications: nrf_desktop: switch to new configuration scheme (#5997)
nrfconnect/sdk-nrf: applications/nrf_desktop/README.rst

The nRF Desktop does not use a single :file:`prj.conf` file.
Configuration files are provided for different build types for each supported board.

> The :file:`prj.conf` represents a ``debug`` build type and is available for each board.

Reviewer: ```suggestion
Each board has its own :file:`prj.conf` file, which represents a ``debug`` build type.
```

> Each other build types has a dedicated file with name following a pattern :file:`prj_${BUILD_TYPE}.conf`.

Reviewer: ```suggestion
Other build types are covered by dedicated files the build type added as a suffix to the ``prj`` part, as per the following list.
For example, the ``release_b0`` build type file name is :file:`prj_release_b0.conf`.
```

> 
> In case there are other configuration files associated with partition layout or child image configuration, these follow the same pattern with build type added to the file name as a suffix.

Reviewer: ```suggestion
If a board has other configuration files, for example associated with partition layout or child image configuration, these follow the same pattern.
```
Note the deleted blank line.

.. include:: /gs_modifying.rst
   :start-after: build_types_overview_start

>    :end-before: build_types_overview_end

Reviewer: These lines describe what a build type is. They need to follow the first paragraph. Move them to line 381.

* ``b0`` -- ``debug`` build type with the support for the B0 bootloader enabled (for :ref:`background DFU <nrf_desktop_bootloader_background_dfu>`).
* ``mcuboot`` -- ``debug`` build type with the support for the MCUboot bootloader enabled (for :ref:`serial recovery DFU <nrf_desktop_bootloader_serial_dfu>` or :ref:`background DFU <nrf_desktop_bootloader_background_dfu>`).
* ``shell`` -- ``debug`` build type with the shell enabled.

In nRF Desktop, not every development kit can support every build type mentioned above.
If the given build type is not supported on the selected DK, an error message will appear when `Building and running`_.
For example, if the ``shell`` build type is not supported on the selected DK, the following notification appears:

.. code-block:: console

   File not found: ./ncs/nrf/applications/nrf_desktop/configuration/nrf52dmouse_nrf52832/prj_shell.conf

|nrf_desktop_build_type_conf|
For example, the nRF52840 Development Kit supports the ``keyboard`` configuration, which is defined in the :file:`prj_keyboard.conf` file in the :file:`configuration/nrf52840dk_nrf52840` directory.
This configuration lets you generate the application with the keyboard role.

> See :ref:`nrf_desktop_porting_guide` for detailed information about the application configuration and how to create build type files for your hardware.

Reviewer: Lines 815-832 need to be updated accordingly with the new build type usage steps, also including the VSC extension.

Reviewer: In other words, we need to update gs_modifying.rst file.

Author: Discussed offline. These documentation changes will be done as a separate PR.

When the bootloader is enabled, the nRF Desktop application uses the partition manager for the layout configuration of the flash memory.
The nRF Desktop configurations with bootloader use static configurations of partitions to ensure that the partition layout will not change between builds.

> Add the :file:`pm_static_${CONF_FILE_BUILD_TYPE}.yml` file to the project's board configuration directory to define the static partition manager configuration for given board and build type.

Author: s/CONF_FILE_BUILD_TYPE/BUILD_TYPE

Author: done

> Add the :file:`pm_static_${CONF_FILE_BUILD_TYPE}.yml` file to the project's board configuration directory to define the static partition manager configuration for given board and build type.

Reviewer: Add an example:
```suggestion
For example, to define the static partition manager for the XYZ board and XYZ board type, you would need to add the :file:`xyz` file.
```

Author: done
